src/helpers/batfish/filters.js

```diff
@@ -0,0 +1,110 @@
+// builds all the data you need to power exampleIndex layout
+// see: https://mapbox.github.io/dr-ui/batfish-helpers/#filters
+const { sortAlpha } = require('./navigation');
+
+function buildFilters(data) {
+  // format page data
+  const pages = data.pages.map((p) => ({
+    path: p.path, // each path is the page's primary key
+    ...p.frontMatter
+  }));
+
+  // find every example index page
+  const exampleIndexPages = pages.filter(
+    (p) => (p.navOrder && p.layout === 'example') || p.layout === 'exampleIndex'
+  );
+
+  // for each example index page, find sub pages and collect uniq filter option arrays
+  return exampleIndexPages.reduce((obj, parent) => {
+    // find the subpages for each example index page
+    const children = pages.filter(
+      (page) => page.path.startsWith(parent.path) && page.path !== parent.path
+    );
+
+    if (children.length > 0) {
+      obj[parent.path] = {
+        topics: generateTopics(children),
+        levels: generateLevels(children),
+        languages: combineArrays(children, 'language'),
+        videos: children.filter((f) => f.video).length > 0, // returns bool
+        products: combineArrays(children, 'products'),
+        pages: pageSorter(children)
+      };
+    }
+    return obj;
+  }, {});
+}
+
+// combines arrays, makes them unique, and sorts them
+function combineArrays(pages, prop) {
+  return [
+    ...new Set(
+      pages
+        .filter((f) => f[prop])
+        .reduce((arr, item) => [...arr, ...item[prop]], [])
+    )
+  ].sort();
+}
+
+// creates an array of unique levels and sorts them
+function generateLevels(pages) {
+  return [...new Set(pages.filter((f) => f.level).map((f) => f.level))].sort();
+}
+
+// creates an array of unique topic(s) and sorts them
+function generateTopics(pages) {
+  const topics = [
+    ...new Set(
+      pages.reduce((arr, page) => {
+        if (page.topics) arr = arr.concat(page.topics);
+        if (page.topic) arr.push(page.topic);
+        return arr;
+      }, [])
+    )
+  ].sort();
+  // check if "Getting started" topic exists
+  const index = topics.indexOf('Getting started');
+  // if "Getting started" exists, splice the array, and move "Getting started" to the front of the list
+  return index > -1 ? [...topics.splice(index, 1), ...topics] : topics;
+}
+
+function hasTopic(page, topic) {
+  return (
+    (page.topic && page.topic === topic) ||
+    (page.topics && page.topics.indexOf(topic) > -1)
+  );
+}
+
+function sortBy(arr, prop) {
+  return arr.sort((a, b) => parseInt(a[prop]) - parseInt(b[prop]));
+}
+
+// sorts pages by "Getting started" topic(s), then by level, then alphabetically by title
```

Comment:
I think I suggested this order, but I'm worried we'll forget why later. I wonder if we should document here in a code comment and/or in the Batfish helpers guide that (even though the comment here is accurate) the _intent_ is to start with "Getting started" examples for product example pages **or** the beginner level tutorials on the tutorials page, then alphabetically by title.

Reply:
Ok! I'll document!